database/mongodb/mongodb_test.go

```diff
@@ -72,26 +72,32 @@ func TestWithAuth(t *testing.T) {
 			if err != nil {
 				t.Fatalf("%v", err)
 			}
-
-			driverWithAuth, err := p.Open(fmt.Sprintf("mongodb://deminem:gogo@%s:%v/testMigration", i.Host(), i.Port()))
-			if err != nil {
-				t.Fatalf("%v", err)
+			testcases := []struct {
+				name            string
+				connectUri      string
+				isErrorExpected bool
+			}{
+				{"right auth data", "mongodb://deminem:gogo@%s:%v/testMigration", false},
+				{"wrong auth data", "mongodb://wrong:auth@%s:%v/testMigration", true},
 			}
-			defer driverWithAuth.Close()
 			insertCMD := []byte(`[{"insert":"hello","documents":[{"wild":"world"}]}]`)
-			err = driverWithAuth.Run(bytes.NewReader(insertCMD))
-			if err != nil {
-				t.Fatalf("%v", err)
-			}
 
-			driverWithWrongAuth, err := p.Open(fmt.Sprintf("mongodb://wrong:auth@%s:%v/testMigration", i.Host(), i.Port()))
-			if err != nil {
-				t.Fatalf("%v", err)
-			}
-			defer driverWithWrongAuth.Close()
-			err = driverWithWrongAuth.Run(bytes.NewReader(insertCMD))
-			if err == nil {
-				t.Fatal("no error with wrong authorization")
+			for _, tcase := range testcases {
+				t.Run(tcase.name, func(t *testing.T) {
+					mc := &Mongo{}
+					d, err := mc.Open(fmt.Sprintf(tcase.connectUri, i.Host(), i.Port()))
+					if err != nil {
```

Comment:
You dont't get error with wrong auth after methods `Connect` and `Ping`. Only after trying to run db commands

Reply:
Oh, that's odd...

It looks like there's an option to authenticate while connecting vs after connecting: https://docs.mongodb.com/manual/tutorial/enable-authentication/#connect-to-the-instance-and-authenticate-as-mytester

Not sure if the Golang driver also supports authentication during connection

Reply:
Yes, for now it's not supporting authenticate during connection

Reply:
In that case, add a note about that in the the test since the behavior isn't intuitive.

It *may* also be worth adding the same to the MongoDB driver README. I'm not sure how user friendly the error messages are...